rust/src/lib.rs

```diff
@@ -184,7 +217,7 @@ pub unsafe extern "C" fn zcashlc_init_data_database(
         match init_wallet_db(&mut db_data, seed) {
             Ok(_) => Ok(0),
             Err(MigratorError::Adapter(WalletMigrationError::SeedRequired)) => Ok(1),
-            Err(e) => Err(format_err!("Error while initializing data DB: {}", e)),
+            Err(e) => Err(e).with_context(|| "Error while initializing data DB"),
```

Comment:
I think this context is being lost in the FFI. We only get "Error while initializing data DB" from the other side. We need to get similar output until we have rich types.

Reply:
Aha, the problem is that I was misled by [the crate's root-level documentation](https://docs.rs/anyhow/1.0.71/anyhow/#details) into believing that the default `Display` impl would show every level. There's [a different page in the documentation](https://docs.rs/anyhow/1.0.71/anyhow/struct.Error.html#display-representations) that states the default `Display` impl (`{}`) only prints the outer-most layer of the error, and we need to use the alternate instead (`{:#}`). However, that isn't accessible via the error's `ToString` implementation, and that is what the `ffi-helpers` crate we are using relies on.

Easy fix: I undo all the changes I thought were me being helpful and structuring the errors more correctly, and just use the `anyhow!()` macro everywhere we were using `format_err!()` before.

Reply:
Put another way, it's not that the context is being lost in the FFI; it's that _only_ the context is being passed across the FFI.